From 070bc8a98e11d741124ba60031a503b722db545d Mon Sep 17 00:00:00 2001
From: Giuseppe Scrivano <gscrivan@redhat.com>
Date: Thu, 3 Apr 2014 15:52:22 +0200
Subject: [PATCH] virtinst, virtualcontroller: support attribute "model"

in the upstream version this is made as part of a much bigger commit.
This change adds only the needed piece and enable virt-manager to
create virtio-scsi controllers.

RHEL-only

Needed by: https://bugzilla.redhat.com/show_bug.cgi?id=1049781

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>
---
 virtinst/VirtualController.py | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/virtinst/VirtualController.py b/virtinst/VirtualController.py
index 8e5d19d..032d91b 100644
--- a/virtinst/VirtualController.py
+++ b/virtinst/VirtualController.py
@@ -1,5 +1,5 @@
 #
-# Copyright 2010  Red Hat, Inc.
+# Copyright 2010, 2014  Red Hat, Inc.
 # Cole Robinson <crobinso@redhat.com>
 #
 # This program is free software; you can redistribute it and/or modify
@@ -73,6 +73,7 @@ class VirtualController(VirtualDevice.VirtualDevice):
         self._index = 0
         self._ports = None
         self._vectors = None
+        self._model = None
 
     def get_type(self):
         return self._controller_type
@@ -100,6 +101,13 @@ class VirtualController(VirtualDevice.VirtualDevice):
     ports = _xml_property(get_ports, set_ports,
                           xpath="./@ports")
 
+    def get_model(self):
+        return self._model
+    def set_model(self, val):
+        self._model = val
+    model = _xml_property(get_model, set_model,
+                          xpath="./@model")
+
     def _extra_config(self):
         return ""
 
@@ -121,6 +129,12 @@ class VirtualControllerFDC(VirtualController):
 
 class VirtualControllerSCSI(VirtualController):
     _controller_type = VirtualController.CONTROLLER_TYPE_SCSI
+    def _extra_config(self):
+        xml = ""
+        if self.model != None:
+            xml += " model='%s'" % self.model
+
+        return xml
 
 class VirtualControllerSATA(VirtualController):
     _controller_type = VirtualController.CONTROLLER_TYPE_SATA
-- 
1.9.0

